app/scripts/controllers/detect-tokens.js

```diff
@@ -51,6 +56,7 @@ export default class DetectTokensController {
     this.detectedTokens = process.env.TOKEN_DETECTION_V2
       ? this.tokensController?.state.detectedTokens
       : [];
+    this._trackMetaMetricsEvent = trackMetaMetricsEvent;
```

Comment:
May be we can just directly use `trackMetaMetricsEvent` rather than adding `this._trackMetaMetricsEvent`.

Reply:
@jpuri , the method is passed in from the constructor and is invoked outside of the constructor. The only way to make this method available to class methods is by attaching it to the class instance. I think, and could be wrong, that the collapsed code in github made it appear to you (as it did me) that the method was invoked in the constructor, but once you expand the code or pull it locally you'll see it is not invoked in the constructor.